src/client/src/region.rs

```diff
@@ -45,7 +46,13 @@ pub struct RegionRequester {
 impl Datanode for RegionRequester {
     async fn handle(&self, request: RegionRequest) -> MetaResult<AffectedRows> {
         self.handle_inner(request).await.map_err(|err| {
-            if matches!(err, RegionServer { .. }) {
+            if !matches!(
+                err,
+                RegionServer {
+                    code: Code::InvalidArgument,
+                    ..
+                }
+            ) {
```

Comment:
If `Code::InvalidArgument`, I think we need to avoid retrying because any request with invalid options consistently fail.

Let's say that we have a distributed system without S3.

If this process isn't added, the system will retry something like the following query:
```sql
CREATE TABLE not_supported_table_storage_option (
  ~
  ts TIMESTAMP NOT NULL DEFAULT current_timestamp(),
  TIME INDEX (ts),
)
PARTITION BY RANGE COLUMNS (id) (
    PARTITION r0 VALUES LESS THAN (5),
   ~
    PARTITION r2 VALUES LESS THAN (MAXVALUE),
)
~
WITH(
  storage = 'S3'
);
```

In meta terminal:
```bash
~
2023-11-11T07:20:39.648723Z ERROR common_procedure::local::runner: Failed to execute procedure metasrv-procedure::CreateTable-a4b4600d-f242-433e-bf7b-f81d04e9a7dd, retry: true err=0: Procedure exec failed
1: Retry later
2: Retry later
3: Failed to request RegionServer, code: Client specified an invalid argument
4: Object store not found: gcs
2023-11-11T07:20:39.649334Z  INFO common_procedure::local::runner: Procedure metasrv-procedure::CreateTable-a4b4600d-f242-433e-bf7b-f81d04e9a7dd retry for the 2 times after 1000 millis
request: RegionRequest { header: Some(RegionRequestHeader { trace_id: 0, span_id: 0, dbname: "" }), body: Some(Create(CreateRequest { region_id: 4398046511104, engine: "mito", column_defs: [RegionColumnDef { column_def: Some(ColumnDef { name: "host", data_type: String, is_nullable: true, default_constraint: [], 
~
1: Retry later
2: Retry later
3: Failed to request RegionServer, code: Client specified an invalid argument
4: Object store not found: s3
```

Reply:
👍  Good catch!

Reply:
Could we wrap this into a method (`should_retry()`) in the `Error` type?